Patch to enable GDAL 3.8.0 build with libxml2 2.12, as noted in
https://lists.osgeo.org/pipermail/gdal-dev/2023-November/057965.html

--- gcore/gdaljp2metadatagenerator.cpp.orig
+++ gcore/gdaljp2metadatagenerator.cpp
@@ -357,7 +357,12 @@
 /************************************************************************/
 
 static void GDALGMLJP2XPathErrorHandler(void * /* userData */,
-                                        xmlErrorPtr error)
+#if LIBXML_VERSION >= 21200
+                                        const xmlError *error
+#else
+                                        xmlErrorPtr error
+#endif
+)
 {
     if (error->domain == XML_FROM_XPATH && error->str1 != nullptr &&
         error->int1 < static_cast<int>(strlen(error->str1)))


--- port/cpl_xml_validate.cpp.orig
+++ port/cpl_xml_validate.cpp
@@ -914,7 +914,7 @@
 
     if (strstr(pszStr, "since this namespace was already imported") == nullptr)
     {
-        xmlErrorPtr pErrorPtr = xmlGetLastError();
+        const xmlError *pErrorPtr = xmlGetLastError();
         const char *pszFilename = static_cast<char *>(ctx);
         char *pszStrDup = CPLStrdup(pszStr);
         int nLen = static_cast<int>(strlen(pszStrDup));
